### PR TITLE
Able to specify subdirectory within the volume to store TiKV data

### LINKED ISSUE #1525: Provide a more convenient way to modify tikv data directory
## Feature Request

**Is your feature request related to a problem? Please describe:**

I want to test TiKV with [Chaos Mesh](https://github.com/pingcap/chaos-mesh), IOChaos requires that the data directory of the application in the target pod should be a subdirectory of `PersistentVolumes`.  
In TiKV, the default data directory is `/var/lib/tikv` not a subdirectory of `PersistentVolumes`. 

**Describe the feature you'd like:**

1. Can we change the default data directory of tikv to "/var/lib/tikv/data" ?
2. Provide a more convenient way to modify tikv data directory.

**Teachability, Documentation, Adoption, Migration Strategy:**
 

1. For some users, they need to modify tikv data directory.
2. Sometimes, we can reuse the `PersistentVolumes` of tikv, eg: store log files in this pv

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 
fixes #1525 
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Add `.tikv.dataSubDir` field to specify subdirectory within the data volume to store TiKV data 
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -349,6 +349,15 @@ type TiKVSpec struct {
...
 	StorageClassName *string `json:"storageClassName,omitempty"`
 
+	// Subdirectory within the volume to store TiKV Data. By default, the data
+	// is stored in the root directory of volume which is mounted at
+	// /var/lib/tikv.
+	// Specifying this will change the data directory to a subdirectory, e.g.
+	// /var/lib/tikv/data if you set the value to "data".
+	// Defaults to "" (volume's root).
+	// +optional
+	DataSubDir string `json:"dataSubDir,omitempty"`
```
Comment (weekface) [inline pkg/apis/pingcap/v1alpha1/types.go:359]:
```suggestion
	DataSubDir *string `json:"dataSubDir,omitempty"`
```

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/types.go:359]: not necessary, as the empty string is the default value

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/types.go:359]: we don't need to distinguish empty value from nil value here

**Comment by cofyc** [comment]
I will add support for PD too.

**Comment by cofyc** [comment]
> * Please note that old cluster should not set this `dataSubDir` which will cause "data lost"
> * PD needs this `dataSubDir` too

done

@Yisaer @weekface PTAL

`TestRenderTiKVStartScript` is moved to a new test file `template_test.go` and add a new test function `TestRenderPDStartScript`.

File: examples/advanced/tidb-cluster.yaml [hunk 2]
```diff
@@ -47,10 +47,20 @@ spec:
       cpu: "100m"
       storage: 1Gi
 
-    ## The storageClassName of the persistent volume for TiKV data storage.
+    ## The storageClassName of the persistent volume for PD data storage.
```
Comment (cofyc) [inline examples/advanced/tidb-cluster.yaml:50]: typo, this is for PD

**Pull request merged**